### PR TITLE
Fix partition entity interest

### PR DESCRIPTION
**Pull request by m-samiec**
#### Description
The test debug interface was broken as the interest was set to only update on the server worker entity, which is no longer the case in v15. This interest now updates correctly on the partition entity.

#### Tests
Tests pass

#### Primary reviewers
@jacquese @ImprobableNic

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp [hunk 1]
```diff
@@ -475,22 +475,27 @@ void USpatialSender::CreateEntityWithRetries(Worker_EntityId EntityId, FString E
...
+		NetDriver->InterestFactory
+			->CreatePartitionInterest(NetDriver->LoadBalanceStrategy, VirtualId, NetDriver->DebugCtx != nullptr /*bDebug*/)
 			.CreateInterestUpdate();
 
-	Connection->SendComponentUpdate(NetDriver->WorkerEntityId, &InterestUpdate);
+	Connection->SendComponentUpdate(PartitionId, &InterestUpdate);
 
-	// Also update the position of the worker entity to the centre of the load balancing region.
-	SendPositionUpdate(NetDriver->WorkerEntityId, NetDriver->LoadBalanceStrategy->GetWorkerEntityPosition());
+	// Also update the position of the partition entity to the center of the load balancing region.
```
Comment (alastairdglennie) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp:497]: Get that Australian spelling outta here

File: SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp [hunk 2]
```diff
@@ -430,7 +430,7 @@ TArray<FWorkerComponentData> EntityFactory::CreateTombstoneEntityComponents(AAct
 TArray<FWorkerComponentData> EntityFactory::CreatePartitionEntityComponents(const Worker_EntityId EntityId,
 																			const InterestFactory* InterestFactory,
 																			const UAbstractLBStrategy* LbStrategy,
-																			VirtualWorkerId VirtualWorker)
+																			VirtualWorkerId VirtualWorker, bool bDebugContexValid)
```
Comment (alastairdglennie) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp:433]: clang-format keeps this one line?

File: SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp [hunk 3]
```diff
@@ -545,13 +555,28 @@ QueryConstraint InterestFactory::CreateAlwaysInterestedConstraint(const AActor*
...
 
-QueryConstraint InterestFactory::CreateAlwaysRelevantConstraint() const
+SpatialGDK::QueryConstraint InterestFactory::CreateGDKSnapshotEntitiesConstraint() const
 {
 	QueryConstraint AlwaysRelevantConstraint;
 
 	Worker_ComponentId ComponentIds[] = { SpatialConstants::STARTUP_ACTOR_MANAGER_COMPONENT_ID,
-										  SpatialConstants::VIRTUAL_WORKER_TRANSLATION_COMPONENT_ID,
-										  SpatialConstants::ALWAYS_RELEVANT_COMPONENT_ID };
+										  SpatialConstants::VIRTUAL_WORKER_TRANSLATION_COMPONENT_ID };
```
Comment (alastairdglennie) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp:563]: Even you don't need it, worth adding one for the player spawner so we don't trip on it later on?

File: SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp [hunk 4]
```diff
@@ -545,13 +555,28 @@ QueryConstraint InterestFactory::CreateAlwaysInterestedConstraint(const AActor*
 	return AlwaysInterestedConstraint;
 }
 
-QueryConstraint InterestFactory::CreateAlwaysRelevantConstraint() const
+SpatialGDK::QueryConstraint InterestFactory::CreateGDKSnapshotEntitiesConstraint() const
 {
 	QueryConstraint AlwaysRelevantConstraint;
```
Comment (alastairdglennie) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp:560]:
```suggestion
	QueryConstraint SnapshotEntitiesConstraint;
```

File: SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp [hunk 5]
```diff
@@ -545,13 +555,28 @@ QueryConstraint InterestFactory::CreateAlwaysInterestedConstraint(const AActor*
...
 	Worker_ComponentId ComponentIds[] = { SpatialConstants::STARTUP_ACTOR_MANAGER_COMPONENT_ID,
-										  SpatialConstants::VIRTUAL_WORKER_TRANSLATION_COMPONENT_ID,
-										  SpatialConstants::ALWAYS_RELEVANT_COMPONENT_ID };
+										  SpatialConstants::VIRTUAL_WORKER_TRANSLATION_COMPONENT_ID };
+
+	for (Worker_ComponentId ComponentId : ComponentIds)
+	{
+		QueryConstraint Constraint;
+		Constraint.ComponentConstraint = ComponentId;
+		AlwaysRelevantConstraint.OrConstraint.Add(Constraint);
```
Comment (alastairdglennie) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp:569]:
```suggestion
		SnapshotEntitiesConstraint.OrConstraint.Add(Constraint);
```

File: SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp [hunk 6]
```diff
@@ -545,13 +555,28 @@ QueryConstraint InterestFactory::CreateAlwaysInterestedConstraint(const AActor*
...
+	}
+
+	return AlwaysRelevantConstraint;
```
Comment (alastairdglennie) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp:572]:
```suggestion
	return SnapshotEntitiesConstraint;
```

**Review by alastairdglennie — APPROVED** [review]
few minors, lgtm

File: SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp (hunk 2 above)
Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp:433]: it's a good line
also nit: 
```suggestion
																			VirtualWorkerId VirtualWorker, bool bDebugContextValid)
```

File: SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp [hunk 7]
```diff
@@ -439,9 +439,10 @@ TArray<FWorkerComponentData> EntityFactory::CreatePartitionEntityComponents(cons
 	TArray<FWorkerComponentData> Components;
 	Components.Add(Position().CreatePositionData());
 	Components.Add(Metadata(FString::Format(TEXT("PartitionEntity:{0}"), { VirtualWorker })).CreateMetadataData());
-	Components.Add(InterestFactory->CreatePartitionInterest(LbStrategy, VirtualWorker).CreateInterestData());
+	Components.Add(InterestFactory->CreatePartitionInterest(LbStrategy, VirtualWorker, bDebugContexValid).CreateInterestData());
```
Comment (jacquese) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp:442]: also here

File: SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp [hunk 8]
```diff
@@ -110,47 +110,57 @@ Interest InterestFactory::CreateServerWorkerInterest(const UAbstractLBStrategy*
...
-Interest InterestFactory::CreatePartitionInterest(const UAbstractLBStrategy* LBStrategy, VirtualWorkerId VirtualWorker) const
+Interest InterestFactory::CreatePartitionInterest(const UAbstractLBStrategy* LBStrategy, VirtualWorkerId VirtualWorker, bool bDebug) const
 {
 	// Add load balancing query
 	Interest PartitionInterest{};
+	Query PartitionQuery{};
+
 	AddLoadBalancingInterestQuery(LBStrategy, VirtualWorker, PartitionInterest);
+
+	// Ensure server worker receives AlwaysRelevant entities.
```
Comment (jacquese) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp:143]: Should this be on the server worker entity?

File: SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp [hunk 9]
```diff
@@ -110,47 +110,57 @@ Interest InterestFactory::CreateServerWorkerInterest(const UAbstractLBStrategy*
...
+	AddComponentQueryPairToInterestComponent(PartitionInterest, SpatialConstants::WELL_KNOWN_COMPONENT_SET_ID, PartitionQuery);
+
+	// Add a self query for completeness
+	PartitionQuery = Query();
+	PartitionQuery.ResultComponentIds = { SpatialConstants::GDK_KNOWN_ENTITY_TAG_COMPONENT_ID };
+	PartitionQuery.Constraint.bSelfConstraint = true;
+	AddComponentQueryPairToInterestComponent(PartitionInterest, SpatialConstants::WELL_KNOWN_COMPONENT_SET_ID, PartitionQuery);
+
+	// Query to know about all the actors tagged with a debug component
+	if (bDebug)
```
Comment (jacquese) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp:156]: I think this one should also be on the server worker entity- I would expect only one of this and the always relevant query to be made from each worker

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp (hunk 1 above)
Comment (jacquese) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp:497]: prefer keep american spelling please for that sweet consistency (is this in a style guide somewhere? I have no idea if this is consistent throughout the GDK)

**Review by jacquese — APPROVED** [review]
couple comments, looks good, mainly concerned about consistent spelling

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp (hunk 1 above)
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp:497]: It's not in the GDK coding guidelines, but is in our documentation guidelines.

File: SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp (hunk 7 above)
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/EntityFactory.cpp:442]:
```suggestion
	Components.Add(InterestFactory->CreatePartitionInterest(LbStrategy, VirtualWorker, bDebugContextValid).CreateInterestData());
```

File: SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp [hunk 10]
```diff
@@ -545,13 +555,28 @@ QueryConstraint InterestFactory::CreateAlwaysInterestedConstraint(const AActor*
 	return AlwaysInterestedConstraint;
 }
 
-QueryConstraint InterestFactory::CreateAlwaysRelevantConstraint() const
+SpatialGDK::QueryConstraint InterestFactory::CreateGDKSnapshotEntitiesConstraint() const
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp:558]:
```suggestion
QueryConstraint InterestFactory::CreateGDKSnapshotEntitiesConstraint() const
```

File: SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp (hunk 8 above)
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp:143]: The server worker entity is used for declaring interest in non-unreal type entities, where as the partition entity gives you interest on those. That's also why the debug component is added to the partition entity interest also. I don't want to receive unreal actor-entities until the partition component interest kicks in. Happy to chat about this though,

**Review by ImprobableNic — APPROVED** [review]

**Pull request closed without merging**